src/Microsoft.DotNet.ProjectModel/Files/PatternGroup.cs

```diff
@@ -12,6 +12,8 @@ namespace Microsoft.DotNet.ProjectModel.Files
 {
     public class PatternGroup
     {
+        private static readonly Dictionary<string, IEnumerable<string>> s_resolvedFilesCache = new Dictionary<string, IEnumerable<string>>();
```

Comment:
Taking a look at this, it will break the project model server, but from what I'm seeing this change won't improve perf unless it's a static cache because `IncludeFilesResolver.GetIncludeFilesCore` always news up a `Matcher`